osu.Game.Tests/Visual/Gameplay/TestSceneStoryboardSamplePlayback.cs

```diff
@@ -31,42 +41,107 @@ private void load(OsuConfigManager config)
+        {
+            AddStep("setup mod", () =>
+            {
+                ModRateAdjust testedMod = (ModRateAdjust)Activator.CreateInstance(expectedMod).AsNonNull();
+                testedMod.SpeedChange.Value = expectedRate;
+                storyboardMods = new[] { testedMod };
+            });
+
+            createPlayerTest();
+            skipIntro();
+
+            AddAssert("sample playback rate matches mod rates", () => allStoryboardSamples.All(sound =>
+                sound.ChildrenOfType<DrawableSample>().First().AggregateFrequency.Value == expectedRate));
+        }
+
+        [TestCase(typeof(ModWindUp), 0.5, 2)]
+        [TestCase(typeof(ModWindUp), 1.51, 2)]
+        [TestCase(typeof(ModWindDown), 2, 0.5)]
+        [TestCase(typeof(ModWindDown), 0.99, 0.5)]
+        public void TestStoryboardSamplesPlaybackWithTimeRampMods(Type expectedMod, double initialRate, double finalRate)
+        {
+            AddStep("setup mod", () =>
+            {
+                ModTimeRamp testedMod = (ModTimeRamp)Activator.CreateInstance(expectedMod).AsNonNull();
+                testedMod.InitialRate.Value = initialRate;
+                testedMod.FinalRate.Value = finalRate;
+                storyboardMods = new[] { testedMod };
+            });
+
+            createPlayerTest();
+            skipIntro();
 
-            AddStep("skip intro", () => InputManager.Key(osuTK.Input.Key.Space));
-            AddAssert("all storyboard samples stopped immediately", () => allStoryboardSamples.All(sound => !sound.IsPlaying));
+            ModTimeRamp gameplayMod = null;
 
-            AddUntilStep("any storyboard samples playing after skip", () => allStoryboardSamples.Any(sound => sound.IsPlaying));
+            AddUntilStep("mod speed change updated", () =>
+            {
+                gameplayMod = Player.GameplayState.Mods.OfType<ModTimeRamp>().Single();
+                return gameplayMod.SpeedChange.Value != initialRate;
+            });
+
+            AddAssert("sample playback rate matches mod rates", () => allStoryboardSamples.All(sound =>
+                sound.ChildrenOfType<DrawableSample>().First().AggregateFrequency.Value == gameplayMod.SpeedChange.Value));
         }
 
-        private void checkForFirstSamplePlayback()
+        private void createPlayerTest()
         {
+            CreateTest(null);
+
             AddAssert("storyboard loaded", () => Player.Beatmap.Value.Storyboard != null);
-            AddUntilStep("any storyboard samples playing", () => allStoryboardSamples.Any(sound => sound.IsPlaying));
+            waitUntilStoryboardSamplesPlay();
         }
 
+        private void waitUntilStoryboardSamplesPlay() => AddUntilStep("any storyboard samples playing", () => allStoryboardSamples.Any(sound => sound.IsPlaying));
+
+        private void allStoryobardSamplesStopped() => AddAssert("all storyboard samples stopped immediately", () => allStoryboardSamples.All(sound => !sound.IsPlaying));
```

Comment:
typo in name

```suggestion
        private void allStoryboardSamplesStopped() => AddAssert("all storyboard samples stopped immediately", () => allStoryboardSamples.All(sound => !sound.IsPlaying));
```

(+ updating usages)